### PR TITLE
Feature controls

### PR DESCRIPTION
**Pull request by MunWolf**
Changed the whole control scheme and added better physics to our entities courtesy of @G4m3M4ni4c

### CODE REVIEW

**Comment by MunWolf** [comment]
Closes #14 and #23

**Comment by MunWolf** [comment]
Testable here https://wolfgaming.github.io/asteroids-advanced/ rotational and directional thrust values are up for debate, I feel like they are a bit too slow to get up to speed and I find the max velocity too high, opinions?

**Comment by JonathanHHenson** [comment]
Could you make the ship fire lasers when you left click please?

**Comment by MunWolf** [comment]
Added.

File: entities/ship.js [hunk 1]
```diff
@@ -11,33 +11,56 @@ function Ship(world, params) {
...
   };
 
-  this.setInputs = function(thrust, rotateleft, rotateright, laser) {
-    inputs.thrust = thrust;
-    inputs.rotateleft = rotateleft;
-    inputs.rotateright = rotateright;
+  this.setInputs = function(targetPoint, thrustForward, thrustBackwards, thrustLeft, thrustRight, stabilizationToggle, laser) {
+    inputs.thrustVector = createVector(
+      (thrustForward ? this.thrustPower.forward : 0) + (thrustBackwards ? -this.thrustPower.backward : 0),
+      (thrustRight ? this.thrustPower.right : 0) + (thrustLeft ? -this.thrustPower.left : 0)
```
Comment (JonathanHHenson) [inline entities/ship.js]: How about this to make strafing easier:
```
var upRight = p5.Vector.dot(p5.Vector.fromAngle(this.heading), createVector(0, -1));
```
```
(upRight ? 1 : -1) * (thrustRight ? this.thrustPower.right : 0) + (thrustLeft ? -this.thrustPower.left : 0)
```

**Comment by JonathanHHenson** [comment]
Looks good now

**Pull request merged**